src/app/scheme/scheme.service.ts

```diff
@@ -432,63 +458,222 @@ export class SchemeService extends ISchemeService {
     }
 
-    getHelp(): Observable<Help[]>
-    {
+    getHelp(): Observable<Help[]> {
         const url = `/api/v2/scheme/${this.scheme.id}/help/`;
         return this.http.get<Help[]>(url).catch((err: HttpErrorResponse) => of([]));
     }
 
-  exportExcel(conf: ExportConfig, path?: string): Observable<HttpResponse<Blob>> {
-    if (!path) {
-      path = 'excel';
-    }
+    exportExcel(conf: ExportConfig, path?: string): Observable<HttpResponse<Blob>> {
+        if (!path) {
+            path = 'excel';
+        }
 
-    const url = `/export/${path}/?scheme_id=${this.scheme.id}`;
-    const opts = {
-      headers: new HttpHeaders({ 'Content-Type': 'application/json' }),
-      observe: 'response' as 'response',
-      responseType: 'blob' as 'blob'
-    };
+        const url = `/export/${path}/?scheme_id=${this.scheme.id}`;
+        const opts = {
+            headers: new HttpHeaders({'Content-Type': 'application/json'}),
+            observe: 'response' as 'response',
+            responseType: 'blob' as 'blob'
+        };
 
-    console.log(conf);
+        console.log(conf);
 
-    return this.http.post(url, conf, opts).pipe(
-      tap(_ => this.log('Export sucessfull')),
-      catchError(this.handleError<HttpResponse<Blob>>('exportExcel', undefined))
-    );
-  }
+        return this.http.post(url, conf, opts).pipe(
+            tap(_ => this.log('Export sucessfull')),
+            catchError(this.handleError<HttpResponse<Blob>>('exportExcel', undefined))
+        );
+    }
 
-    copy(scheme_id: number, dry_run: boolean): Observable<any>
-    {
+    copy(scheme_id: number, dry_run: boolean): Observable<any> {
         const url = `/api/v2/scheme/${this.scheme.id}/copy/`;
-        return this.http.post<any>(url, { scheme_id, dry_run }).pipe(
-            catchError((err: any): Observable<boolean> =>
-            {
+        return this.http.post<any>(url, {scheme_id, dry_run}).pipe(
+            catchError((err: any): Observable<boolean> => {
                 alert(err.error + '\n' + err.message);
                 return of(undefined);
             }));
     }
 
-    public modify_structure<T>(settingName: string, data: (T | { id: number })[]) {
+    private static getSchemeFieldKey(setting: Structure_Type): keyof Scheme_Detail {
+        const keys: Record<Structure_Type, keyof Scheme_Detail> = {
+            [Structure_Type.ST_DEVICE]: 'device',
+            [Structure_Type.ST_SECTION]: 'section',
+            [Structure_Type.ST_DEVICE_ITEM_TYPE]: 'device_item_type',
+            [Structure_Type.ST_DIG_PARAM_TYPE]: 'dig_param_type',
+            [Structure_Type.ST_DIG_TYPE]: 'dig_type',
+            [Structure_Type.ST_DIG_STATUS_TYPE]: 'dig_status_type',
+            [Structure_Type.ST_DIG_STATUS_CATEGORY]: 'dig_status_category',
+            [Structure_Type.ST_DIG_MODE_TYPE]: 'dig_mode_type',
+            [Structure_Type.ST_VALUE_VIEW]: 'value_view',
+        } as any;
+        return keys[setting];
+    }
```

Comment:
Чот не могу понять, имена же одинаковые, зачем тут продублировано?
```suggestion
    private static getSchemeFieldKey(setting: Structure_Type): keyof Scheme_Detail {
        const keys: Structure_Type[] = [
            Structure_Type.ST_DEVICE, 
            Structure_Type.ST_SECTION,
            Structure_Type.ST_DEVICE_ITEM_TYPE,
            Structure_Type.ST_DIG_PARAM_TYPE,
            Structure_Type.ST_DIG_TYPE,
            Structure_Type.ST_DIG_STATUS_TYPE,
            Structure_Type.ST_DIG_STATUS_CATEGORY,
            Structure_Type.ST_DIG_MODE_TYPE,
            Structure_Type.ST_VALUE_VIEW,
        ];
        return keys.includes(setting) ? settings : undefined;
    }
```

Reply:
Это имело смысл, т.к. запросы могли не совпадать с полями схемы.